### PR TITLE
feat: Serialize charge receipts

### PR DESCRIPTION
**Pull request by bemwen**
## Description

The purpose of this PR is to add serialization of Charge Receipts to CIM/XML.

This PR:
* Adds serialization of charge receipts to CIM/XML
* Moves ReceiptStatus and ReasonCode domain objects to shared AvailableData
* Moves ReceiptStatus mapping to live with MarketDocument so it can be shared between various documents
* Moves ReasonCode mapping to live with MarketActivityRecord so it can be shared between various documents
* Adds charge receipt as a document type

## References

* #823

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/AvailableChargeReceiptData/AvailableChargeReceiptData.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+using System.Collections.Generic;
+using GreenEnergyHub.Charges.Domain.AvailableChargeLinkReceiptData;
+using GreenEnergyHub.Charges.Domain.AvailableData;
+using GreenEnergyHub.Charges.Domain.MarketParticipants;
+using NodaTime;
+
+namespace GreenEnergyHub.Charges.Domain.AvailableChargeReceiptData
+{
+    /// <summary>
+    /// All data necessary for one activity records in a charge link receipt
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/AvailableChargeReceiptData/AvailableChargeReceiptData.cs:25]: One or many activity record(s)?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/ChargeReceiptBundle/Cim/CimChargeReceiptConstants.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace GreenEnergyHub.Charges.Infrastructure.ChargeReceiptBundle.Cim
+{
+    /// <summary>
+    /// Strings used in CIM/XML for elements, namespaces or attributes that we need to
+    /// use when parsing and generating a XML document
+    ///
+    /// This class is responsible for string that are used in the document area of the XML
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/ChargeReceiptBundle/Cim/CimChargeReceiptConstants.cs:21]:
```suggestion
    /// This class is responsible for strings that are used in the document area of the XML
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/ChargeReceiptBundle/Cim/CimChargeReceiptConstants.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    /// which is specific to charge link messages
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/ChargeReceiptBundle/Cim/CimChargeReceiptConstants.cs:22]:
```suggestion
    /// which is specific to charge messages
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/ChargeReceiptBundle/Cim/ChargeReceiptCimSerializerTests.cs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            for (var i = 1; i <= noOfReasons; i++)
+            {
+                var text = string.Empty;
+                if (i % 2 == 0)
+                {
+                    text = "Text" + no + "_" + i;
+                }
+
+                reasonCodes.Add(new AvailableChargeReceiptDataReasonCode(
+                    ReasonCode.IncorrectChargeInformation,
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/ChargeReceiptBundle/Cim/ChargeReceiptCimSerializerTests.cs:165]: Is it ok for the testing purposet, that it is always `IncorrectChargeInformation`?

**Review by HenrikSommer — APPROVED** [review]

**Review by Mech0z — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/AvailableChargeReceiptData/AvailableChargeReceiptData.cs (hunk 1 above)
Comment (bemwen) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/AvailableChargeReceiptData/AvailableChargeReceiptData.cs:25]: One of these is one receipt. The bundle part comes later

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/ChargeReceiptBundle/Cim/ChargeReceiptCimSerializerTests.cs (hunk 4 above)
Comment (bemwen) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/ChargeReceiptBundle/Cim/ChargeReceiptCimSerializerTests.cs:165]: Until we add validation yes, we do not want to try to guess other codes we will need

**Pull request merged**